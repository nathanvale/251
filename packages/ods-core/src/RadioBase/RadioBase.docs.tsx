/* eslint-disable react/display-name */
import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { getSwitchBaseDocs } from "../_private/components/AbstractSwitch/switch-docs";
import { RadioBase, RadioBaseProps } from "./RadioBase";

export const docs: ComponentDocs<RadioBaseProps> = {
  category: "Internal",
  componentName: "RadioBase",
  description: [
    "This is a basic radio component without any labels or helper text.",
    "Use this atomic component to build more complicated components.",
    "The Radio components uses this under the hood.",
  ].join(" "),
  propDescriptions: getSwitchBaseDocs("radio"),
  migrationGuide: false,
  examples: [
    {
      label: "Uncontrolled",
      description: "Simple use case of an uncontrolled radio",
      Code: () => <RadioBase id="tc-radio" />,
    },
    {
      label: "Controlled",
      description: "It becomes controlled when checked prop is provided",
      Code: () => <RadioBase id="tc-radio" checked />,
    },
    {
      label: "Error",
      description: "Turns red when it is in error state and it is not checked",
      Code: () => <RadioBase id="tc-radio" error />,
    },
    {
      label: "Disabled",
      description: "All turn grey when it is disabled",
      Code: () => <RadioBase id="tc-radio" disabled checked />,
    },
  ],
  snippets: [
    {
      label: "Uncontrolled",
      Code: () => <RadioBase id="radio1" />,
    },
  ],
};
