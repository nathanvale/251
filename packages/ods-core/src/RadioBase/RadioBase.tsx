import React from "react";
import {
  RadioProps as MuiRadioProps,
  Radio as MuiRadio,
} from "@material-ui/core";
import { BaseSwitchBaseProps } from "@origin-digital/ods-types";
import { AbstractSwitchBase } from "../_private/components/AbstractSwitch/AbstractSwitchBase";

export interface RadioBaseProps extends BaseSwitchBaseProps {
  muiProps?: MuiRadioProps;
}

export const RadioBase = (props: RadioBaseProps) => {
  return <AbstractSwitchBase<MuiRadioProps> Component={MuiRadio} {...props} />;
};

RadioBase.displayName = "RadioBase";
RadioBase.defaultProps = {};
