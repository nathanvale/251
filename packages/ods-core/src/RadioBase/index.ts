export { RadioBase } from "./RadioBase";
export type RadioBaseProps = import("./RadioBase").RadioBaseProps;
