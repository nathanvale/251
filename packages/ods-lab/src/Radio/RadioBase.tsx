import * as React from "react";
import MuiRadio from "@material-ui/core/Radio";
import makeStyles from "@material-ui/styles/makeStyles";
import { AllColors } from "../AllColors/AllColors";
import { IRadioProps } from "./Radio";
import { FormControlLabel, FormHelperText, Typography } from "..";

export interface RadioBaseProps extends IRadioProps {
  "data-id"?: string;
  onChange?: (e: React.ChangeEvent<any>, checked: boolean) => void;
}

interface InputWrapperProps {
  children: React.ReactNode;
  border?: boolean;
  checked?: boolean;
}

const newStyles = makeStyles({
  inputWrapper: {
    border: (props: InputWrapperProps) => {
      return props.border ? `1px solid ${AllColors.grey56}` : "none";
    },
    borderColor: (props: InputWrapperProps) => {
      return props.border ? `1px solid ${AllColors.lightOrange}` : "none";
    },
    borderRadius: (props: InputWrapperProps) => {
      return props.border ? 10 : 0;
    },
    padding: (props: InputWrapperProps) => {
      return props.border ? 12 : 0;
    },
    marginBottom: (props: InputWrapperProps) => {
      return props.border ? 8 : 0;
    },
  },
  inlineWrapper: {
    display: "inline-flex",
    alignItems: "center",
    justifyContent: "space-between",
    width: "100%",
  },
});

const InputWrapper = (props: InputWrapperProps) => {
  const styles = newStyles(props);
  return <div className={styles.inputWrapper}>{props.children}</div>;
};

interface InlineWrapperProps {
  children: React.ReactNode;
}

const InlineWrapper = (props: InlineWrapperProps) => {
  const styles = newStyles(props);
  return <div className={styles.inlineWrapper}>{props.children}</div>;
};

export const RadioBase = (props: RadioBaseProps) => {
  const {
    "data-id": dataId,
    disabled,
    value,
    color,
    label,
    labelPlacement,
    helpText,
    checked,
    onChange,
    name,
    border,
    description,
    error,
  } = props;
  return (
    <InputWrapper border={border} checked={checked}>
      <InlineWrapper>
        <FormControlLabel
          key={value}
          data-id={dataId}
          disabled={disabled}
          value={value}
          checked={checked}
          onChange={onChange}
          color={color ? color : "default"}
          name={name}
          control={
            <MuiRadio
              inputProps={{ "aria-label": props.label }}
              color={color ? color : "default"}
            />
          }
          labelPlacement={labelPlacement}
          label={label}
        />
        {description && (
          <Typography variant="caption">{description}</Typography>
        )}
      </InlineWrapper>
      {helpText && <FormHelperText error={error}>{helpText}</FormHelperText>}
    </InputWrapper>
  );
};
