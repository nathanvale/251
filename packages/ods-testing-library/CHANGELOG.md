# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.3.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-testing-library@0.2.0...@origin-digital/ods-testing-library@0.3.0) (2020-02-12)


### Features

* **tests, columns:** alignY default changed to top ([c23c964](https://bitbucket.orgn.io/od/origin-ui/commits/c23c964))





# 0.2.0 (2020-01-31)


### Features

* set up jest for snapshots ([6e6fbfa](https://bitbucket.orgn.io/od/origin-ui/commits/6e6fbfa))
