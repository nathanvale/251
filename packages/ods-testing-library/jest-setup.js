import { configure } from "@testing-library/react";

require("jest-styled-components");

configure({ testIdAttribute: "data-id" });
