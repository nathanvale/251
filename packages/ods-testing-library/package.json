{
  "name": "@origin-digital/ods-testing-library",
  "description": "ODS testing tools that enhance react-testing-library",
  "license": "UNLICENSED",
  "author": "Nathan Vale <nathan.vale@origin.com.au>",
  "version": "0.5.0",
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "files": [
    "dist",
    "jest-setup.js"
  ],
  "scripts": {
    "build": "yarn build:cjs && yarn build:es",
    "build:cjs": "BUILD_FORMAT=cjs origin-scripts build --monorepo --delete-dir-on-start --out-dir dist/cjs",
    "build:es": "BUILD_WEBPACK=true origin-scripts build --monorepo --out-dir dist/es"
  },
  "sideEffects": false,
  "typings": "dist/typings/index.d.ts",
  "dependencies": {
    "@babel/runtime-corejs3": "^7.8.3",
    "@testing-library/jest-dom": "^5.1.1"
  },
  "peerDependencies": {
    "@origin-digital/ods-themes": "*",
    "@testing-library/react": "9.x",
    "jest-styled-components": "6.x",
    "react": "^16.9.0"
  },
  "devDependencies": {
    "@testing-library/react": "9.x"
  },
  "publishConfig": {
    "registry": "https://artifactory.origindigital-dac.com.au/artifactory/api/npm/npm-internal/"
  }
}
