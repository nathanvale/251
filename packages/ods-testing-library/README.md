# @origin-digital/ods-testing-library

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->

- [Installation](#installation)
- [Why do I need this package?](#why-do-i-need-this-package)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

This package enhances the render function of `react-testing-library` so that the `<OriginThemeProvider/>` is wrapped around your rendered component.

All other API features of `react-testing-library` are also included.

## Installation

Install the package in your project directory with:

```sh
yarn add @origin-digital/ods-testing-library
```

This package also has a peer dependency on the following:

```sh
// with yarn
yarn add @origin-digital/ods-core
yarn add -D @testing-library/react
yarn add -D jest-styled-components
```

## Why do I need this package?

**The problem**

When using react-testing-library and ODS with the jest testing framework it is necessary to wrap **all** your components in our `<OriginThemeProvider/>` - otherwise the react component cannot be rendered and tested.

**The solution**

`@origin-digital/ods-testing-library` creates a wrapper around the `react-testing-library` render function. It is here that we wrap the `<OriginThemeProvider/>`. All other API features of `react-testing-library` are also included.

This practice is recommended by the `react-testing-library` documentation itself...

https://testing-library.com/docs/native-testing-library/setup#custom-render

The origin-ui dogfoods this package itself but it can also be used in your own app to test by adding to your jest config:

```
{
...,
setupFilesAfterEnv: [@origin-digital/ods-testing-library/jest-setup],
...
}
```

and usage in your tests:

```
import React from 'react';
import {render} from '@origin-digital/ods-testing-library';
import {Box} from '../Box/Box';
import {Stack} from './Stack';

test('It can stack', () => {
  const {container} = render(
    <Stack>
      <Box backgroundColor="red" padding="medium" />
      <Box backgroundColor="orange" padding="medium" />
      <Box backgroundColor="blue" padding="medium" />
    </Stack>,
  );
  expect(container.firstChild).toMatchSnapshot();
});
```
