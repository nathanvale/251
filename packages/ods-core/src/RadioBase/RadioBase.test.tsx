import React from "react";
import { queryByAttribute, render } from "@origin-digital/ods-testing-library";
import { RadioBase } from "./RadioBase";

test("it renders an unchecked RadioBase", () => {
  const { container } = render(<RadioBase id="radioInput" />);

  expect(container).toMatchSnapshot();
});

test("it renders a checked RadioBase", () => {
  const { container } = render(<RadioBase id="radioInput" checked />);

  expect(container).toMatchSnapshot();
});

test("it renders a RadioBase in error", () => {
  const { container } = render(<RadioBase id="radioInput" error />);

  expect(container).toMatchSnapshot();
});

test("it renders a disabled RadioBase", () => {
  const { container } = render(<RadioBase id="radioInput" disabled />);

  expect(container).toMatchSnapshot();
});

describe("uncontrolled RadioBase", () => {
  test("it passes down id and data-id", () => {
    const { container } = render(
      <RadioBase data-id="radioRoot" id="myRadio" />,
    );

    const chkRoot = queryByAttribute("data-id", container, "radioRoot");
    const input = queryByAttribute("id", container, "myRadio");

    expect(chkRoot).toBeDefined();
    expect(input?.tagName.toLowerCase()).toEqual("input");
  });

  test("it is clicked, it will set Mui-Checked prop", () => {
    const { container } = render(
      <RadioBase data-id="radioRoot" id="myRadio" />,
    );

    const chkRoot = queryByAttribute("data-id", container, "radioRoot");
    const input = queryByAttribute("id", container, "myRadio");

    expect(chkRoot).not.toHaveClass("Mui-checked");

    input?.click();

    expect(chkRoot).toHaveClass("Mui-checked");
  });
});
