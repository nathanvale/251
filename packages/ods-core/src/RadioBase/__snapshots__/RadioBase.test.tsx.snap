// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a RadioBase in error 1`] = `
<div>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-21 MuiRadio-root AbstractSwitchBase-error-root-19 AbstractSwitchBase-error-root-20 MuiRadio-colorSecondary MuiIconButton-colorSecondary"
    data-id="radioInput"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        class="PrivateSwitchBase-input-24"
        id="radioInput"
        type="radio"
        value=""
      />
      <div
        class="PrivateRadioButtonIcon-root-25"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-26"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
          />
        </svg>
      </div>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </span>
</div>
`;

exports[`it renders a checked RadioBase 1`] = `
<div>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-12 MuiRadio-root AbstractSwitchBase-root-10 AbstractSwitchBase-root-11 MuiRadio-colorSecondary PrivateSwitchBase-checked-13 Mui-checked MuiIconButton-colorSecondary"
    data-id="radioInput"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        checked=""
        class="PrivateSwitchBase-input-15"
        id="radioInput"
        type="radio"
        value=""
      />
      <div
        class="PrivateRadioButtonIcon-root-16 PrivateRadioButtonIcon-checked-18"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-17"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
          />
        </svg>
      </div>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </span>
</div>
`;

exports[`it renders a disabled RadioBase 1`] = `
<div>
  <span
    aria-disabled="true"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-30 MuiRadio-root AbstractSwitchBase-root-28 AbstractSwitchBase-root-29 MuiRadio-colorSecondary PrivateSwitchBase-disabled-32 Mui-disabled MuiIconButton-colorSecondary Mui-disabled Mui-disabled"
    data-id="radioInput"
    tabindex="-1"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        class="PrivateSwitchBase-input-33"
        disabled=""
        id="radioInput"
        type="radio"
        value=""
      />
      <div
        class="PrivateRadioButtonIcon-root-34"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-35"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
          />
        </svg>
      </div>
    </span>
  </span>
</div>
`;

exports[`it renders an unchecked RadioBase 1`] = `
<div>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiRadio-root AbstractSwitchBase-root-1 AbstractSwitchBase-root-2 MuiRadio-colorSecondary MuiIconButton-colorSecondary"
    data-id="radioInput"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        class="PrivateSwitchBase-input-6"
        id="radioInput"
        type="radio"
        value=""
      />
      <div
        class="PrivateRadioButtonIcon-root-7"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-8"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
          />
        </svg>
      </div>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </span>
</div>
`;
